src/controllers/profile.controller.js

```diff
@@ -0,0 +1,131 @@
+import { ApiError } from '../exceptions/api.error.js';
+import { userService } from '../services/user.service.js';
+import { User } from '../models/user.js';
+import bcrypt from 'bcrypt';
+import { validatePassword } from '../utils/methods.js';
+import { emailService } from '../services/email.services.js';
+import jwt from 'jsonwebtoken';
+
+const rename = async (req, res) => {
+  const { newName, email } = req.body;
+
+  if (!newName) {
+    throw ApiError.badRequest('Bad request, need to provide new name of user');
+  }
+
+  const user = await userService.getByEmail(email);
+
+  if (!user) {
+    throw ApiError.badRequest('user not found');
+  }
+
+  await User.update({ name: newName }, { where: { id: user.id } });
+
+  res.status(200).send('Done');
+};
+
+const changePassword = async (req, res) => {
+  const { email, password, newPassword, confirmationPassword } = req.body;
+  const user = await userService.getByEmail(email);
+
+  if (!user) {
+    throw ApiError.badRequest('user not found');
+  }
+
+  if (!email || !password || !newPassword || !confirmationPassword) {
+    throw ApiError.badRequest('Need to provide all fields');
+  }
+
+  const isPasswordRight = await bcrypt.compare(password, user.password);
+
+  if (!isPasswordRight) {
+    throw ApiError.badRequest('Password is not right.');
```

Comment:
The error message 'Password is not right.' could be improved to be more user-friendly, such as 'Incorrect password.'